pkg/api/validation/validation.go

```diff
@@ -372,6 +372,11 @@ func ValidateObjectMetaUpdate(newMeta, oldMeta *api.ObjectMeta, fldPath *field.P
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("resourceVersion"), newMeta.ResourceVersion, "must be specified for an update"))
 	}
 
+	// Generation should be the same or at most incremented by one.
+	if newMeta.Generation != oldMeta.Generation && newMeta.Generation != oldMeta.Generation+1 {
```

Comment:
> Besides, for Deployment, changing it wouldn't have any effect because it wouldn't change podTemplateHash.

Right, I have a pull downstream where the apiserver identifies generation increments for DeploymentConfigs from clients and translates them to version bumps in order to kick a new deployment of the same thing. How would you imagine a redeployment of the same thing for Deployments?